Source/Wikitivity.PreInstallInstanceSetting/PreInstallEventHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Net;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using kCura.EventHandler;
+using kCura.EventHandler.CustomAttributes;
+using kCura.Relativity.Client;
+using Relativity.API;
+using Relativity.Services.Interfaces.InstanceSetting;
+using Relativity.Services.Interfaces.InstanceSetting.Model;
+using Relativity.Services.Objects;
+
+namespace Wikitivity.PreInstallInstanceSetting
+{
+	[RunOnce(true)]
+	[RunTarget(kCura.EventHandler.Helper.RunTargets.Workspace)]
+	[kCura.EventHandler.CustomAttributes.Description("Create Wikitivity WebAPI Instance Setting")]
+	[System.Runtime.InteropServices.Guid("b7918424-59a0-4374-af45-17e033270213")]
+	public class PreInstallEventHandler : kCura.EventHandler.PreInstallEventHandler
+	{
+		public override Response Execute()
+		{
+			IAPILog logger = Helper.GetLoggerFactory().GetLogger();
+			logger.LogVerbose("Log information throughout execution.");
+			// Update Security Protocol
+			ServicePointManager.SecurityProtocol = SecurityProtocolType.Ssl3 | SecurityProtocolType.Tls | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls12;
+
+			//Construct a response object with default values.
+			kCura.EventHandler.Response retVal = new kCura.EventHandler.Response();
+
+			try
+			{
+
+				var proxy = Helper.GetServicesManager().CreateProxy<IInstanceSettingManager>(ExecutionIdentity.System);
+				int operationResults = CreateWikitivityWebAPIInstanceSetting(proxy, retVal).Result;

```

Comment:
Before creating the instance key, I would first query to see if it exists and only create the instance setting it doesn't already exist.